src/lib/field.ts

```diff
@@ -0,0 +1,1194 @@
+    return this.toBigInt().toString();
+  }
+
+  /**
+   * Assert that this {@link Field} is equal another "field-like" value.
+   * Calling this function is equivalent to `Field(...).equals(...).assertEquals(Bool(true))`.
+   * See {@link Field.equals} for more details.
+   *
+   * **Important**: If an assertion fails, the code throws an error.
+   *
+   * @param value - the "field-like" value to compare & assert with this {@link Field}.
+   * @param message? - a string error message to print if the assertion fails, optional.
+   */
+  assertEquals(y: Field | bigint | number | string, message?: string) {
+    try {
+      if (this.isConstant() && isConstant(y)) {
+        if (this.toBigInt() !== toFp(y)) {
+          throw Error(`Field.assertEquals(): ${this} != ${y}`);
+        }
+        return;
+      }
+      Snarky.field.assertEqual(this.value, Field.#toVar(y));
+    } catch (err) {
+      throw withMessage(err, message);
+    }
+  }
+
+  /**
+   * Add a "field-like" value to this {@link Field} element.
+   *
+   * @example
+   * ```ts
+   * const x = Field(3);
+   * const sum = x.add(5);
+   *
+   * sum.assertEquals(Field(8));
+   * ```
+   *
+   * **Warning**: This is a modular addition in the pasta field.
+   * @example
+   * ```ts
+   * const x = Field(1);
+   * const sum = x.add(Field(-7));
+   *
+   * // If you try to print sum - `console.log(sum.toBigInt())` - you will realize that it prints a very big integer because this is modular arithmetic, and 1 + (-7) circles around the field to become p - 6.
+   * // You can use the reverse operation of addition (substraction) to prove the sum is calculated correctly.
+   *
+   * sum.sub(x).assertEquals(Field(-7));
+   * sum.sub(Field(-7)).assertEquals(x);
+   * ```
+   *
+   * @param value - a "field-like" value to add to the {@link Field}.
+   *
+   * @return A {@link Field} element equivalent to the modular addition of the two value.
+   */
+  add(y: Field | bigint | number | string): Field {
+    if (this.isConstant() && isConstant(y)) {
+      return new Field(Fp.add(this.toBigInt(), toFp(y)));
+    }
+    // return new AST node Add(x, y)
+    let z = Snarky.field.add(this.value, Field.#toVar(y));
+    return new Field(z);
+  }
+
+  /**
+   * Negate a {@link Field}. This is equivalent to multiplying the {@link Field} by -1.
+   *
+   * @example
+   * ```ts
+   * const negOne = Field(1).neg();
+   * negOne.assertEquals(-1);
+   * ```
+   *
+   * @example
+   * ```ts
+   * const someField = Field(42);
+   * someField.neg().assertEquals(someField.mul(Field(-1))); // This statement is always true regardless of the value of `someField`
+   * ```
+   *
+   * **Warning**: This is a modular negation. For details, see the {@link sub} method.
+   *
+   * @return A {@link Field} element that is equivalent to the element multiplied by -1.
+   */
+  neg() {
+    if (this.isConstant()) {
+      return new Field(Fp.negate(this.toBigInt()));
+    }
+    // return new AST node Scale(-1, x)
```

Comment:
did you mean the flip the parameters in the comment? Should we flip them in the actual representation to match this?

Reply:
I'm flipping them in the `Snarky.field.scale` function to match the type!